components/emc2101.rst

```diff
@@ -0,0 +1,177 @@
+EMC2101 Fan Controller and Temperature sensor
+=============================================
+
+.. seo::
+    :description: Instructions for setting up EMC2101 fan controller and temperature sensor in ESPHome.
+    :image: emc2101.jpg
+    :keywords: EMC2101
+
+The ``emc2101`` component allows you to control a cooling fan from ESPHome (`datasheet <https://ww1.microchip.com/downloads/aemDocuments/documents/MSLD/ProductDocuments/DataSheets/EMC2101-Data-Sheet-DS20006703.pdf>`__). It also provides temperature
+sensors and a tachometer. The fan can be controlled either by a PWN signal or by outputing a DC voltage
+using its DAC.
+
+The :ref:`I²C Bus <i2c>` is required to be set up in your configuration for this sensor to work.
+
+.. figure:: images/emc2101-full.png
+    :align: center
+    :width: 50.0%
+
+    EMC2101 Fan Controller and Temperature sensor
+
+.. note::
+
+    Original image at https://learn.adafruit.com/assets/97827 created by 
+    by user `Bryan Siepert <https://learn.adafruit.com/u/siddacious>`__ under
+    `Creative Commons BY-SA 3.0 <https://creativecommons.org/licenses/by-sa/3.0/>`__ and modified
+    for this documentation.
+
+
+
+EMC2101 Component
+-----------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    i2c:
+      sda: GPIO8
+      scl: GPIO9
+
+    emc2101:
+      pwm:
+        resolution: 8
+
+    output:
+      - platform: emc2101
+        id: fan_duty_cycle
+
+    fan:
+      - platform: speed
+        name: "Hall Cabinet Fan"
+        output: fan_duty_cycle
+        speed_count: 16
+
+    sensor:
+      - platform: emc2101
+        internal_temperature:
+          id: internal_temperature_sensor
+          name: "Internal Temperature Sensor"
+        speed:
+          id: speed_sensor
+          name: "Speed Sensor"
+        duty_cycle:
+          id: duty_cycle_sensor
+          name: "Duty Cycle Sensor"
+        update_interval: 5s
```

Comment:
This main example should only apply to the main component. 
`sensor` and `output` have their own yaml sections below and we dont need to repeat them here.

If you want to have a full example, please make it the last item on the page in its own section.


```suggestion
.. code-block:: yaml

    # Example configuration entry
    emc2101:
      pwm:
        resolution: 8
```